roboteam_ai/src/coach/heuristics/CoachHeuristics.cpp

```diff
@@ -43,7 +44,7 @@ double CoachHeuristics::calculatePassLineScore(const Vector2 &position, const Wo
 double CoachHeuristics::getClosestOpponentAngleToPassLine(const Vector2 &position, const Ball &ball,
         double smallestAngle) {
 
-    for (const auto &robot : world::world->getAllRobots()) {
+    for (const auto &robot : world::world->getThem()) {
```

Comment:
This is inconsistent with the rest of the world calls, sometimes you call world::world->them, sometimes you use a copy of the world. I suggest you create one copy of the world without robots in the OffensiveScore::calculateOffensivePositionScore() and PassScore::calculatePassScore() functions and just pass that on as const reference to it's children functions so you're guaranteed to work with the same world in all the functions, and you only have to do things like removing our robots once.